src/modules/dashboard/components/CreatePaymentDialog/CreatePaymentDialog.tsx

```diff
@@ -0,0 +1,127 @@
+
+import { pipe, mapPayload, withKey } from '~utils/actions';
+import { Address } from '~types/index';
+import { ActionTypes } from '~redux/index';
+import Dialog from '~core/Dialog';
+import { ActionForm } from '~core/Fields';
+import { SpinnerLoader } from '~core/Preloaders';
+import {
+  useColonyQuery,
+  ColonySubscribedUsersDocument,
+  useLoggedInUser,
+  useUser,
+} from '~data/index';
+
+import DialogForm from './CreatePaymentDialogForm';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+
+export interface FormValues {
+  fromDomain?: string;
+  toAssignee?: string;
+  amount: string;
+  tokenAddress?: Address;
+  reason?: string;
+}
+
+interface Props {
+  cancel: () => void;
+  close: (params: object) => void;
+  colonyAddress: Address;
+}
+
+const CreatePaymentDialog = ({ colonyAddress, cancel, close }: Props) => {
+  const validationSchema = yup.object().shape({
+    fromDomain: yup.number().required(),
+    toAssignee: yup.number().required(),
+    amount: yup.string().required(),
+    tokenAddress: yup.string().required(),
+    reason: yup.string().max(90),
+  });
+
+  const { walletAddress: loggedInUserWalletAddress } = useLoggedInUser();
+
+  const loggedInUser = useUser(loggedInUserWalletAddress);
+
+  const { data: colonyData } = useColonyQuery({
+    variables: { address: colonyAddress },
+  });
+
+  const { data: subscribedUsersData } = useQuery(
+    ColonySubscribedUsersDocument,
+    { variables: { colonyAddress } },
+  );
+
+  const tokens = (colonyData && colonyData.colony.tokens) || [];
+  const nativeTokenAddress = colonyData && colonyData.colony.nativeTokenAddress;
+
+  const transform = useCallback(
+    pipe(
+      mapPayload((payload) => {
+        // Find the selected token's decimals
+        const selectedToken = tokens.find(
+          (token) => token.address === payload.tokenAddress,
+        );
+        const decimals = getTokenDecimalsWithFallback(
+          selectedToken && selectedToken.decimals,
+        );
+
+        // Convert amount string with decimals to BigInt (eth to wei)
+        const amount = bigNumberify(moveDecimal(payload.amount, decimals));
+
+        return {
+          ...payload,
+          colonyAddress,
+          amount,
+          fromDomain: parseInt(payload.fromDomain, 10),
+          toAssignee: parseInt(payload.toAssignee, 10),
+        };
+      }),
+      withKey(colonyAddress),
+    ),
+    [colonyAddress, tokens],
+  );
+
+  return (
+    <ActionForm
+      initialValues={{
+        fromDomain: ROOT_DOMAIN_ID.toString(),
+        toAssignee: loggedInUser,
+        amount: '',
+        tokenAddress: nativeTokenAddress,
+        reason: '',
+      }}
+      validationSchema={validationSchema}
+      submit={ActionTypes.MOVE_FUNDS_BETWEEN_POTS}
+      error={ActionTypes.MOVE_FUNDS_BETWEEN_POTS_ERROR}
+      // Close dialog immediately to give way for GasStation
+      success={ActionTypes.MOVE_FUNDS_BETWEEN_POTS}
```

Comment:
This are not really relevant here, so please remove them, plus the success action is wrong.

But I'll deal with them when I get to wiring this up.

Reply:
`submit`, `error`, and `success` can't be undefined. Is there a placeholder ActionType that I can use for now?

Reply:
Nah. Leave them like they are, I'll pull this in after you merge and I'll put the proper ones in